src/state/Behavior.ts

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2025 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+Please see LICENSE in the repository root for full details.
+*/
+
+import { BehaviorSubject, Observable } from "rxjs";
+
+import { type ObservableScope } from "./ObservableScope";
+
+/**
+ * A stateful, read-only reactive value. As an Observable, it is "hot" and
+ * always replays the current value upon subscription.
+ *
+ * A Behavior is to BehaviorSubject what Observable is to Subject; it does not
+ * provide a way to imperatively set new values. For more info on the
+ * distinction between Behaviors and Observables, see
+ * https://monoid.dk/post/behaviors-and-streams-why-both/.
+ */
+export type Behavior<T> = Omit<BehaviorSubject<T>, "next" | "observers">;
+
+/**
+ * Creates a Behavior which never changes in value.
+ */
+export function constant<T>(value: T): Behavior<T> {
+  return new BehaviorSubject(value);
+}
+
+declare module "rxjs" {
+  interface Observable<T> {
+    /**
+     * Converts this Observable into a Behavior. This requires the Observable to
+     * synchronously emit an initial value.
+     */
+    behavior(scope: ObservableScope): Behavior<T>;
+  }
+}
```

Comment:
I am wondering if we want two api's one where the observable provides the current/initial value and one where we can pass in an initial value manually so that also non state observables can be used as behaviors?

Reply:
Yeah, having a shortcut for `observable.pipe(startWith(initialValue)).behavior(scope)` wouldn't be a bad idea. I could see it making sense to either overload the `Observable.behavior` method, add a separate `behaviorFrom(scope, initialValue)` method, or call this thing `stepper` like other FRP libraries do.